### PR TITLE
added documentation for `dataType` for `bind` symbols

### PR DESCRIPTION
**Pull request by vlada-shubina**
### Solution
Documentation for https://github.com/dotnet/templating/pull/5822

### CODE REVIEW

File: docs/Binding-and-project-context-evaluation.md [hunk 1]
```diff
@@ -21,14 +21,14 @@ The priority of the sources are following:
...
-
+|Name|Description|Mandatory|
+|---|---|---|
+|`type`|`bind`|yes|
+|`binding`| Mandatory. The name of the source and parameter in the source to take the value from. The syntax follows: `<source prefix>:<parameter name>`.|yes|
+|`replaces`|The text to be replaced by the symbol value in the template files content.|no|
+|`fileRename`|The portion of template filenames to be replaced by the symbol value.|no| 	 
+|`defaultValue`|The value assigned to the symbol if no value was provided from external source(s). Recommended to be used when `replaces` and/or `fileRename` is used. In case default value is not specified and no value was provided from external source(s) the replacement won't be performed. |no|
+|`dataType`|The value assigned to the symbol if no value was provided from external source(s). Allowed values are: "bool", "float", "int", "hex", "text", "string". If not specified, the value type will be inferred. In case the type of values might be ambiguous, consider specifying the desired datatype for processing. In case the conversion of value to the type fails, the symbol will be skipped from  further processing. |no|
```
Comment (JanKrivanek) [inline docs/Binding-and-project-context-evaluation.md:31]: Should we link to the symbol data types section https://github.com/dotnet/templating/blob/main/docs/Reference-for-template.json.md#parameter-symbol?

It has a bit more detail. The only difference is the support for choice type .... thinking about it a bit - I gues we can and should support it for bind symbols as well What do you think?

**Review by JanKrivanek — APPROVED** [review]
Looks good.

I'm thinking whether we should unify parameter symbols and bind symbols resolution completely (support for choices, `OnParameterError` removal). But that would be a different work

File: docs/Binding-and-project-context-evaluation.md (hunk 1 above)
Comment (vlada-shubina) [reply docs/Binding-and-project-context-evaluation.md:31]: I agree. Not only for `bind` symbols, but for all other cases where we support `dataType`.

Comment (vlada-shubina) [reply docs/Binding-and-project-context-evaluation.md]: Created: https://github.com/dotnet/templating/issues/5828

**Pull request merged**